File: pset6/caesar.py
import sys

if not(len(sys.argv) == 2):
    print("Erorr\n")
    exit(1)
    
#modify a key for more comfortable usage
if int(sys.argv[1]) <= 26:
    key = int(sys.argv[1])
else:
    key = int(sys.argv[1])%26
    
text = input("plaintext: ")

#encrypt text
for i in range(len(text)):
    if(text[i].isalpha()): 
        if (text[i].isupper()):
            text = text.replace(text[i], chr((((ord(text[i])-64)+key)%26)+64), 1)
        else:
            text = text.replace(text[i], chr((((ord(text[i])-96)+key)%26)+96), 1)

print("ciphertext: ", text)

File: pset2/caesar.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>
#include <cs50.h>
#include <ctype.h>

//function that cipher text
char *crypt(char *text, int key);

int main(int argc, string argv[])
{
    if (argc!=2)
    {
        printf("Erorr\n");
        return 1;
    }
    
    //modify a key for more comfortable usage
    int key = atoi(argv[1]) <= 26 ? (atoi(argv[1])):(atoi(argv[1])%26);
    
    printf("plaintext: ");
    string text = GetString();

    printf("ciphertext: %s\n", crypt(text, key));
    
    return 0;
}

char *crypt(char *text, int key)
{
    
    for (int i = 0, len = strlen(text); i < len; i++)
    {
        if(isalpha(text[i])) 
        {
            if (isupper(text[i]))
            {
                text[i] = (((text[i]-64)+key)%26)+64;
            }
            else
            {
                text[i] =  (((text[i]-96)+key)%26)+96;
            }
        }
    }
    return text;
}